android/modules/ui/src/java/ti/modules/titanium/ui/widget/webview/TiUIWebView.java

```diff
@@ -931,4 +931,11 @@ protected void disableHWAcceleration()
 	{
 		Log.d(TAG, "Do not disable HW acceleration for WebView.", Log.DEBUG_MODE);
 	}
+
+	public void setMediaPlaybackRequiresUserGesture(boolean enabled)
+	{
+		if (Build.VERSION.SDK_INT >= 17) {
+			getWebView().getSettings().setMediaPlaybackRequiresUserGesture(enabled);
+		}
+	}
```

Comment:
This causes an "opcode" warning in the Android log on API Level 16 and lower. While harmless, we don't want to add any more noise to the log. Doing the following would be better...
`
	public void setMediaPlaybackRequiresUserGesture(boolean enabled)
	{
		if (Build.VERSION.SDK_INT >= 17) {
			TiUIWebView webView = getWebView();
			if (webView != null) {
				ApiLevel17.setMediaPlaybackRequiresUserGesture(webView.getSettings(), enabled);
			}
		}
	}

	private static class ApiLevel17
	{
		private ApiLevel17() {}

		public static boolean setMediaPlaybackRequiresUserGesture(
			android.webkit.WebSettings settings, boolean enabled)
		{
			if (settings != null) {
				settings.setMediaPlaybackRequiresUserGesture(enabled);
			}
		}
	}
`

The above solution is based on Google's blog post "How to have your (Cup)cake and eat it too".  It takes advantage of Java's lazy class loading to avoid the opcode warning. Google's Android Supports library does something very similar to this.